os/fs/smartfs/smartfs_utils.c

```diff
@@ -2931,12 +2984,12 @@ static int smartfs_set_transaction(struct smartfs_mountpt_s *fs, uint16_t sector
 		ret = FS_IOCTL(fs, BIOC_WRITESECT, (unsigned long)&req);
 		if (ret != OK) {
 			fdbg("Writing failed %u %u\n", req.logsector, req.offset);
-			return ERROR;
+			return ret;
```

Comment:
Same

Reply:
Yes, by returning "ret", we can know the error type of "FS_IOCTL" in detail.